controllers/vspherevm_controller.go

```diff
@@ -252,6 +264,25 @@ func (r vmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.R
 	// Always issue a patch when exiting this function so changes to the
 	// resource are patched back to the API server.
 	defer func() {
+		// Before computing ready condition, make sure that VirtualMachineProvisioned is always set.
+		// NOTE: This is required because v1beta2 conditions comply to guideline requiring conditions to be set at the
+		// first reconcile.
+		if c := v1beta2conditions.Get(vmContext.VSphereVM, infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition); c != nil {
+			if vmContext.VSphereVM.Status.Ready {
+				v1beta2conditions.Set(vmContext.VSphereVM, metav1.Condition{
+					Type:   infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition,
+					Status: metav1.ConditionTrue,
+					Reason: infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Reason,
+				})
+			} else {
+				v1beta2conditions.Set(vmContext.VSphereVM, metav1.Condition{
+					Type:   infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition,
+					Status: metav1.ConditionFalse,
+					Reason: infrav1.VSphereMachineVirtualMachineNotProvisionedV1Beta2Reason,
+				})
+			}
+		}
```

Comment:
This should use VSphereVMVirtualMachineProvisionedV1Beta2Condition, right? (instead of VSphereMachine* condition & reasons)

Reply:
Nice catch, thanks!